backend/v3/storage/database/query.go

```diff
@@ -92,6 +99,29 @@ const (
 	OrderDirectionDesc
 )
 
+type ResultLock uint8
+
+const (
+	ResultLockNone ResultLock = iota
+	// ResultLockForUpdate locks the selected rows for update.
+	// Use this lock if you do not update the primary key of the selected rows.
+	// See [FOR NO KEY UPDATE](https://www.postgresql.org/docs/current/explicit-locking.html#LOCKING-ROWS) for additional information.
+	// This prevents other transactions from modifying or deleting the rows
+	// until the current transaction is committed or rolled back.
+	ResultLockForUpdate
+	// ResultLockForPrimaryKeyUpdate locks the selected rows for update.
+	// Use this lock if you update the primary key of the selected rows.
+	// See [FOR UPDATE](https://www.postgresql.org/docs/current/explicit-locking.html#LOCKING-ROWS) for additional information.
+	// This prevents other transactions from modifying or deleting the rows
+	// until the current transaction is committed or rolled back.
+	ResultLockForPrimaryKeyUpdate
```

Comment:
Reading the docs on postgresql.org , I understand that this is more generally for columns being used in indices or foreign keys. Hence, I believe the name is misleading ?

I would personally keep the same naming as Postgres.

The other doubt I have is if this should be here or in the `dialect/postgres` package. I don't know if these type of locks are specific to PG or are SQL standards.

Reply:
The SQL standard allows `FOR UPDATE` so i change it to only allow `FOR UPDATE` instead of the other ones